### PR TITLE
Change string append to use d3-path

### PR DESCRIPTION
**Pull request by Sebastien-Ahkrin**
Closes: https://github.com/cheminfo/nmrium/issues/1351

### CODE REVIEW

File: src/component/2d/tools/slicing/HorizontalSliceChart.tsx [hunk 1]
```diff
@@ -31,16 +32,17 @@ function HorizontalSliceChart({
...
-        '',
-      );
-      return path;
+      const pathBuilder = path();
+
+      pathBuilder.moveTo(scaleX(pathPoints.x[0]), scaleY(pathPoints.y[0]));
+      for (let i = 1; i < pathPoints.x.length; i++) {
+        pathBuilder.lineTo(
+          scaleX(pathPoints.x[i]),
+          scaleY(pathPoints.y[i + 1]),
```
Comment (targos) [inline src/component/2d/tools/slicing/HorizontalSliceChart.tsx:41]:
```suggestion
          scaleY(pathPoints.y[i]),
```

**Review by targos — CHANGES_REQUESTED** [review]
Can you implement the more efficient variant [as discussed in the issue](https://github.com/cheminfo/nmrium/issues/1351#issuecomment-1024873783) ?

File: src/component/utility/PathBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  public toString() {
+    return this.array.join(' ');
+  }
+
+  /**
+   * Add space before and after the string
+   */
+  private format(element: string) {
+    return ` ${element} `;
```
Comment (targos) [inline src/component/utility/PathBuilder.ts:20]: The join `' '` is not enough?

File: src/component/2d/Left1DChart.tsx [hunk 3]
```diff
@@ -40,18 +41,15 @@ function Left1DChart({
...
-          return accumulator;
-        },
-        '',
-      );
-      return path;
+
+      const pathBuilder = new PathBuilder();
+      pathBuilder.moveTo(scaleY(pathPoints.y[lastYIndex]), scaleX(lastXIndex));
+
+      for (let i = pathPoints.x.length - 2; i > 0; i--) {
```
Comment (lpatiny) [inline src/component/2d/Left1DChart.tsx:48]: Pretty sure it should be `i>=0`

@Sebastien-Ahkrin Could you check this.

File: src/component/2d/Left1DChart.tsx [hunk 4]
```diff
@@ -40,18 +41,15 @@ function Left1DChart({
...
-        (accumulator, point, index) => {
-          accumulator += ` L  ${scaleY(pathPoints.y[index])} ${scaleX(point)}`;
-          return accumulator;
-        },
-        '',
-      );
-      return path;
+
+      const pathBuilder = new PathBuilder();
+      pathBuilder.moveTo(scaleY(pathPoints.y[lastYIndex]), scaleX(lastXIndex));
```
Comment (lpatiny) [inline src/component/2d/Left1DChart.tsx:46]: Those 2 variables lastYindex and lastXIndex are rather useless and I would simply but the corresponding `pathPoints.x.length - 1` directly here.

File: src/component/2d/tools/slicing/VerticalSliceChart.tsx [hunk 5]
```diff
@@ -39,18 +40,15 @@ function VerticalSliceChart({
...
-          return accumulator;
-        },
-        '',
-      );
-      return path;
+
+      const pathBuilder = new PathBuilder();
+
+      pathBuilder.moveTo(scaleY(pathPoints.y[lastYIndex]), scaleX(lastXIndex));
+      for (let i = pathPoints.x.length - 2; i > 0; i--) {
```
Comment (lpatiny) [inline src/component/2d/tools/slicing/VerticalSliceChart.tsx:47]: Same as previous comment.

**Review by lpatiny — COMMENTED** [review]
Nice improvement but I think there are still 2 bugs.

**Pull request merged**